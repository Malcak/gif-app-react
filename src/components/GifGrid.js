import React from 'react';
import PropTypes from 'prop-types';
import { useFechGifs } from '../hooks/useFechGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFechGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
