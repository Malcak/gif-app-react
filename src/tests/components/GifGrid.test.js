import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { GifGrid } from '../../components/GifGrid';
import { useFechGifs } from '../../hooks/useFechGifs';
jest.mock('../../hooks/useFechGifs');

describe('tests in <GifGrid /> component', () => {
  const category = 'One punch';
  test('should display the <GifGrid /> component correctly', () => {
    useFechGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should display items when images are loaded', () => {
    const gifs = [
      {
        id: 'a1b2c3',
        title: 'title test',
        url: 'https://localhost/test/test.gif',
      },
      {
        id: '1a2b3c',
        title: 'title test',
        url: 'https://localhost/test/test.gif',
      },
    ];
    useFechGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
