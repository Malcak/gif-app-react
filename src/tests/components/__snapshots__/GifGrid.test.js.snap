// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <GifGrid /> component should display items when images are loaded 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="a1b2c3"
      key="a1b2c3"
      title="title test"
      url="https://localhost/test/test.gif"
    />
    <GifGridItem
      id="1a2b3c"
      key="1a2b3c"
      title="title test"
      url="https://localhost/test/test.gif"
    />
  </div>
</Fragment>
`;

exports[`tests in <GifGrid /> component should display the <GifGrid /> component correctly 1`] = `
<Fragment>
  <h3>
    One punch
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
